### PR TITLE
feat: Support msgpack event via minidump attachment

### PR DESCRIPTION
**Pull request by bruno-garcia**

### CODE REVIEW

File: src/sentry/lang/native/minidump.py [hunk 1]
```diff
@@ -96,6 +105,65 @@ def merge_process_state_event(data, state, cfi=None):
     data.setdefault('debug_meta', {})['images'] = images
 
 
+def merge_attached_event(mpack_event, data):
+    # Merge msgpack serialized event.
+    if sys.getsizeof(mpack_event) > MAX_MSGPACK_EVENT_SIZE_BYTES:
```
Comment (mitsuhiko) [inline src/sentry/lang/native/minidump.py]: just make this `len(mpack_event)` :)

File: src/sentry/lang/native/minidump.py [hunk 2]
```diff
@@ -96,6 +105,65 @@ def merge_process_state_event(data, state, cfi=None):
...
+
+    try:
+        unpacker = msgpack.Unpacker(mpack_breadcrumbs)
+    except UnpackException as e:
+        minidumps_logger.exception(e)
+        return
+
+    levels = {-1: 'debug', 0: 'info', 1: 'warning', 2: 'error', 3: 'critical'}
+    breadcrumbs = []
+    for crumb in unpacker:
```
Comment (mitsuhiko) [inline src/sentry/lang/native/minidump.py]: the try/except for `UnpackException` needs to go around the loop as the iteration can raise an `UnpackException` on bad data.

File: src/sentry/lang/native/minidump.py [hunk 3]
```diff
@@ -96,6 +105,65 @@ def merge_process_state_event(data, state, cfi=None):
...
+    if sys.getsizeof(mpack_breadcrumbs) > MAX_MSGPACK_BREADCRUMB_SIZE_BYTES:
+        return
+
+    try:
+        unpacker = msgpack.Unpacker(mpack_breadcrumbs)
+    except UnpackException as e:
+        minidumps_logger.exception(e)
+        return
+
+    levels = {-1: 'debug', 0: 'info', 1: 'warning', 2: 'error', 3: 'critical'}
```
Comment (mitsuhiko) [inline src/sentry/lang/native/minidump.py]: I would instead have the client send strings instead of integers here.

**Review by mitsuhiko — CHANGES_REQUESTED** [review]

File: src/sentry/lang/native/minidump.py (hunk 1 above)
Comment (bruno-garcia) [reply src/sentry/lang/native/minidump.py]: This one won't work: `TypeError: object of type '_io.BytesIO' has no len()`

**Review by mitsuhiko — APPROVED** [review]

**Pull request merged**